rubicon/objc/types.py

```diff
@@ -54,16 +232,25 @@ def encoding_for_ctype(vartype):
     NSRangeEncoding = b'{_NSRange=II}'
     UIEdgeInsetsEncoding = b'{UIEdgeInsets=ffff}'
     NSEdgeInsetsEncoding = b'{NSEdgeInsets=ffff}'
+    PyObjectEncoding = b'^{_object=i^{_typeobject}}'
 
-NSIntegerEncoding = encoding_for_ctype(NSInteger)
-NSUIntegerEncoding = encoding_for_ctype(NSUInteger)
-CGFloatEncoding = encoding_for_ctype(CGFloat)
+NSIntegerEncoding = ctype_to_encoding(NSInteger)
+NSUIntegerEncoding = ctype_to_encoding(NSUInteger)
+CGFloatEncoding = ctype_to_encoding(CGFloat)
 
 # Special case so that NSImage.initWithCGImage_size_() will work.
 CGImageEncoding = b'{CGImage=}'
 
 NSZoneEncoding = b'{_NSZone=}'
 
+register_preferred_encoding(PyObjectEncoding, py_object)
+
+
+class UnknownPointer(c_void_p):
+    """Placeholder for the b'^?' "unknown pointer" type. Not to be confused with a b'^v' void pointer.
+    Usually a b'^?' is a function pointer, but because the encoding doesn't contain the function signature, you need to manually create a CFUNCTYPE with the proper types, and cast this pointer to it.
+    """
+register_preferred_encoding(b'^?', UnknownPointer)
```

Comment:
Ah - but you could make it a dual-purpose decorator. The simplest implementation would be:
```
def register_encoding(encoding, ctype=None):
    def _register(cls):
        _encoding_to_ctype_map.setdefault(encoding, cls)
        _ctype_to_encoding_map.setdefault(cls, encoding)
        return cls
    if ctype is None:
        return _register
    else:
        _register(ctype)
```
Then in usage:
```
@register_encoding('xxx')
class MyStruct(Structure):
   ...
```
or
```
register_encoding('xxx', MyStruct)
```